### DISCUSSION TITLE
A way to disable challenge categories

### DISCUSSION THREAD

**Discussion by MaSven** [OWASP/wrongsecrets#501] [Ideas]
I try to get this into our organization environment as a way of learning. But some categories are not relevant for us. So maybe a way of defining categories to disable could help here.

**Comment by commjoen** [comment]
Hey @MaSven ! 
thank you so much for wanting to use this! 
Just to make sure: What type of categories do you mean? the "focus" types as visible on the main page its challenge table? (https://wrongsecrets.herokuapp.com/) Or do you mean more the environment related type of challenges? (e.g. no "cloud" or no "k8s" or no "vault" challenges)

**Reply by MaSven** [reply]
The focus types. For example, I want to disable all web3 related challenges.

**Reply by commjoen** [reply]
That is actually a great idea! But now the question would become: how would you like to be able to configure this for yourself?
What would you prefer?
1. a config file with disabled focus-types as part of the container you release
2. an env-var with disabled focus-types
3. ...anything you can think of? We need to make sure it remains within the context of 1 container as we don't like to force people to locally need to do docer-compose, minikube, etc...

Next: in the UI: is it ok if they are just "disabled" with a notification of being disabled similar to what we now do with unsupported challenges due to lack of a given infrastructure being setup?

**Reply by commjoen** [reply]
After deciding on this, we will create an issue out of this so we can pick it up for development :) .

**Reply by MaSven** [reply]
I would prefer the first one. With this, we can provide a forked version of wrongsecrets to the organization for people, who want to run it themselves.

**Reply by commjoen** [reply]
We created #502 out of this and will eventually pick it up :) . 
BTW: Feel free to volunteer on it or if you know anyone who wants to pick it up ^^!

**Comment by commjoen** [comment]
This feature will be implemented with https://github.com/OWASP/wrongsecrets/pull/1083 .

**Comment by commjoen** [comment]
The feature has been implemented. Thank you for the idea!
